scipy/constants/_codata.py

```diff
@@ -1496,40 +1532,163 @@
+        'inverse meter-electron volt relationship': (h * c) / e,
+        'inverse meter-joule relationship': h * c,
+        'inverse meter-kelvin relationship': h * c / k,
+        'inverse meter-kilogram relationship': h / c,
+        'inverse of conductance quantum': 1 / (2 * e**2 / h),
+        'Josephson constant': K_J,
+        'joule-electron volt relationship': 1 / e,
+        'joule-hertz relationship': 1 / h,
+        'joule-inverse meter relationship': 1 / (h * c),
+        'joule-kelvin relationship': 1 / k,
+        'joule-kilogram relationship': 1 / c**2,
+        'kelvin-electron volt relationship': k / e,
+        'kelvin-hertz relationship': k / h,
+        'kelvin-inverse meter relationship': k / (h * c),
+        'kelvin-kilogram relationship': k / c**2,
+        'kilogram-electron volt relationship': c**2 / e,
+        'kilogram-hertz relationship': c**2 / h,
+        'kilogram-inverse meter relationship': c / h,
+        'kilogram-joule relationship': c**2,
+        'kilogram-kelvin relationship': c**2 / k,
+        'Loschmidt constant (273.15 K, 100 kPa)': 100e3/273.15 / k,
+        'Loschmidt constant (273.15 K, 101.325 kPa)': 101.325e3/273.15 / k,
+        'mag. flux quantum': h / (2*e),
+        'molar gas constant': N_A * k,
+        'molar Planck constant': h * N_A,
+        'molar volume of ideal gas (273.15 K, 100 kPa)': N_A * k * 273.15 / 100e3,
+        'molar volume of ideal gas (273.15 K, 101.325 kPa)': N_A * k * 273.15 / 101.325e3,
+        'natural unit of action': (h / (2*pi)),
+        'natural unit of action in eV s': (h / (2*pi))/e,
+        'Planck constant in eV/Hz': h/e,
+        'reduced Planck constant': h / (2*pi),
+        'reduced Planck constant in eV s': (h / (2*pi)) / e,
+        'reduced Planck constant times c in MeV fm': (h / (2*pi)) * c / (e * 1e6 * 1e-15),
+        'second radiation constant': h * c / k,
+        'Stefan-Boltzmann constant': 2 * pi**5 * k**4 / (15 * h**3 * c**2),
+        'von Klitzing constant': R_K,
+        'Wien frequency displacement law constant': alpha_W * k / h,
+        'Wien wavelength displacement law constant': h * c / (x_W * k),
+    }
+    return replace
+
+
+# -----------------------------------------------------------------------------
 
 
-def parse_constants_2002to2014(d: str) -> dict[str, tuple[float, str, float]]:
+def parse_constants_2002to2014(d: str, exact_func) -> dict[str, tuple[float, str, float]]:
     constants = {}
+    exact = {}
+    need_replace = set()
     for line in d.split('\n'):
         name = line[:55].rstrip()
         val = float(line[55:77].replace(' ', '').replace('...', ''))
+        is_truncated = '...' in line[55:77]
+        is_exact = '(exact)' in line[77:99]
+        if is_truncated and is_exact:
+            # missing decimals, use computed exact value
+            need_replace.add(name)
+        elif is_exact:
+            exact[name] = val
+        else:
+            assert not is_truncated
```

Comment:
Here and elsewhere:
Is this for control flow? If yes, could you replace these with proper exceptions? The issue is that assertions are missing if someone use `python -O`